src/nodes.coffee

```diff
@@ -1081,6 +1150,25 @@ exports.Value = class Value extends Base
 
     fragments
 
+  ast: ->
+    ret = @base.ast()
+    for prop, propIndex in @properties
+      ret =
+        mergeAstLocationData(
+          merge {
+            type: 'MemberExpression'
+            object: ret
+            property: prop.ast()
+            computed: prop instanceof Index or prop.name?.unwrap() not instanceof PropertyName
+            optional: !!prop.soak
+            shorthand: !!prop.shorthand
+          }, prop.astLocationData()
+          ret
+        )
+      if propIndex is 0 and @base instanceof Parens and @base.locationData?
```

Comment:
This is a case I ran into when doing some Prettier tests: typically, the location data for the overall `MemberExpression` just needs to span the location of its `object` and its `property` (eg for `a.b` it spans `a`'s location through `b`'s location). But if the `object` is parenthesized, we also need to include the location of the parens in the `MemberExpression`'s data (eg `( a ).b`). Included a generated location data test case